### DISCUSSION TITLE
useStorage with server-side files encoding

### DISCUSSION THREAD

**Discussion by bart** [nitrojs/nitro#1230] [Q&A]
Hey community 👋 

I have a very special problem and maybe you can help.
I'm using the Nitro useStorage feature for assets in Nuxt.js (server-side / api) deployed on Vercel.
When running something like `await useStorage('assets:public').getItem('logo.png')` it returns the string
of that file which is correct. But unfortunately image data is usually not utf8 so what I get when converting it
into a buffer to use it in PDFkit I'm getting an `Unknown image format` error.

Is there any way to set encoding `getItem()` should return data?

Any help is highly appreciated, thanks!

**Comment by maou-shonen** [comment]
[getItemRaw](https://unstorage.unjs.io/usage#getitemrawkey-opts)

**Reply by bart** [reply]
Unfortunately that doesn't work. I'm getting an empty string when using getItemRaw() on a file. Or may I use it wrong?

**Reply by maou-shonen** [reply]
It should return a `Buffer`,
If you find a bug, you can create a issues in unstorage.

**Reply by bart** [reply]
Unfortunately it looks like a bug. I'm getting back a `Buffer` but it's just a small peace of the entire file.

```
<Buffer 6e 75 6c 6c>
```

Will create a bug later, thanks!

**Comment by bart** [comment]
Might be continued here: https://github.com/unjs/unstorage/issues/222
